### PR TITLE
chore(errors): use asserts in place of errors

### PR DESCRIPTION
**Pull request by dtfiedler**
Changes:
- removing the use of `pcall` in happy path tests that we want to throw if they fail
- updates to user `asserts` where we want to throw, standardizing our error handling
- updates to utils tests
- removing some unused functions

PE-7144

### CODE REVIEW

**Review by fedellen — DISMISSED** [review]

**Review by fedellen — APPROVED** [review]

File: src/gar.lua [hunk 1]
```diff
@@ -1243,22 +1141,19 @@ end
...
+	assert(gateway, "Gateway not found")
 
 	-- Only allow modification of the allow list when allowDelegatedStaking is set to false or a current allow list is in place
-	if gateway.settings.allowDelegatedStaking == true and not gateway.settings.allowedDelegatesLookup then
-		error("Allow listing only possible when allowDelegatedStaking is set to 'allowlist'")
-	end
+	assert(
+		gateway.settings.allowDelegatedStaking == true and not gateway.settings.allowedDelegatesLookup,
+		"Allow listing only possible when allowDelegatedStaking is set to 'allowlist'"
+	)
```
Comment (dtfiedler) [inline src/gar.lua:1150]: need to fix

File: src/gar.lua [hunk 2]
```diff
@@ -1295,22 +1190,19 @@ end
...
+	assert(gateway, "Gateway not found")
 
 	-- Only allow modification of the allow list when allowDelegatedStaking is set to false or a current allow list is in place
-	if gateway.settings.allowDelegatedStaking == true or not gateway.settings.allowedDelegatesLookup then
-		error("Allow listing only possible when allowDelegatedStaking is set to 'allowlist'")
-	end
+	assert(
+		gateway.settings.allowDelegatedStaking == true or not gateway.settings.allowedDelegatesLookup,
+		"Allow listing only possible when allowDelegatedStaking is set to 'allowlist'"
+	)
```
Comment (dtfiedler) [inline src/gar.lua:1199]: need to fix and add test

Comment (dtfiedler) [reply src/gar.lua]: https://github.com/ar-io/ar-io-network-process/pull/202/commits/f76447d3fd890ef960259e2552da5a3d5b0b3924

File: src/gar.lua (hunk 1 above)
Comment (dtfiedler) [reply src/gar.lua]: https://github.com/ar-io/ar-io-network-process/pull/202/commits/f76447d3fd890ef960259e2552da5a3d5b0b3924

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**